Makefile

```diff
@@ -237,7 +217,7 @@ protos: $(PROTO_GOS)
 
 GENERATE_FILES ?= true
 
-%.pb.go:
+%.pb.go: %.proto
```

Comment:
This is better than it was before, but it's still not quite right: we won't regenerate `.pb.go` files if a transitive dependency `.proto` has changed.

For example, if `pkg/mimirpb/mimir.proto` changes, we will regenerate `pkg/mimirpb/mimir.pb.go`, but not `pkg/distributor/distributorpb/distributor.pb.go`, even though `pkg/distributor/distributorpb/distributor.proto` imports `mimirpb.proto`. Under most circumstances, this won't be an issue (either there's nothing that needs to change in `distributor.pb.go`, or the Go compiler will catch the problem). 

The safest thing to do here would be to depend on all `.proto` files, but that also feels like overkill. Thoughts?